src/socket_io.cpp

```diff
@@ -105,6 +105,11 @@ namespace libtorrent
 			}
 			*port_pos = '\0';
 			++port_pos;
+			if (*port_pos != ':')
```

Comment:
you dereference the port_pos iterator here, but it's possible that it's the one-past-end by this point, which is undefined behavior.